src/react/README.md

```diff
@@ -54,6 +54,8 @@ At a client-level, changes to the connection status can be subscribed to by prov
 
 Listeners can be provided for events related to a particular `Room` instances status changes via `onRoomStatusChange`. Furthermore, discontinuities in the event stream for room features can be monitored via the `onDiscontinuity` event listener. Room-level events pertain to the nearest parent `RoomProvider` in the React component tree.
 
+Listeners passed in are stored internally to create a stable reference to provide stability to the listener subscription. Therefore, the listener values passed to hooks are **ignored after the first render**. If you have optional functionality in your application, you should handle this within the listener, rather than swapping out listener functions.
```

Comment:
Do we mean that if you

```
// first render
useFeature({listener: () => console.log("batman")});

// second render
useFeature({listener: () => console.log("yoda")});
```

this always prints `batman` on events?

I thought line 48, updating the ref.current in useLayoutCallback, makes sure we actually call the correct listener, so outputting `yoda` for events after second render?

Reply:
Yes that's correct - although we use `useLayoutEffect` we memoize the value on the first render, so we always get the same thing back.